content/docs/upgrading/upgrade.md

```diff
@@ -4,11 +4,67 @@ description = "Upgrading your deployment to a later version of Kubeflow"
 weight = 30
 +++
 
-Until version 1.0, Kubeflow makes no promises of backwards compatibility or 
-upgradeability. There is **no clean upgrade path** to the latest version of 
-Kubeflow ({{% kf-latest-version %}}).
+Upgrading your Kubeflow deployment is supported if your deployment is v0.7.0 or later.
 
-Nonetheless, here are some instructions for updating your deployment:
+Prerequisites:
+
+* Download the latest kfctl binary from the
+  [Kubeflow releases page](https://github.com/kubeflow/kubeflow/releases/tag/{{% kf-latest-version %}}).
+
+* Ensure that your kubeflow namespace is annotated with the following label:
+```
+  labels:
+    control-plane: kubeflow
+```
+* You must have a local Kubeflow application directory matching your current
+   deployment.
+
+Upgrade instructions:
+
+1. Create an upgrade spec in the parent directory of the Kubeflow application. An example is provided
+[here](https://github.com/kubeflow/manifests/blob/v0.7-branch/kfdef/kfctl_upgrade_gcp_iap_0.7.1.yaml).
+
+```yaml
+apiVersion: kfupgrade.apps.kubeflow.org/v1alpha1
+kind: KfUpgrade
+metadata:
+  name: kf-upgrade-v0.7.1
+spec:
+  currentKfDef:
+    # Replace with the name of your Kubeflow app
+    name: kubeflow-app
+    version: v0.7.0
+  newKfDef:
+    # Replace with the name of your kubeflow app
+    name: kubeflow-app
+    version: v0.7.1
+  # Replace this with the path to the KfDef that you are upgrading to
+  baseConfigPath: https://raw.githubusercontent.com/kubeflow/manifests/v0.7-branch/kfdef/kfctl_gcp_iap.0.7.1.yaml
+```
+1. Run the upgrade command:
+
+```
+kfctl apply -f ${UPGRADE_SPEC} -V
+```
+
+1. Alternatively you can run a build command first:
+```
+kfctl build -f ${UPGRADE_SPEC} -V
+```
+This will create a new Kubeflow application in the same directory (the name
```

Comment:
If I understand correctly, then the name isn't something people need to specify themselves, but the word "should" makes me think I should do something about the format of the name.

Would this wording work instead?

```
The `build` command creates a new Kubeflow application in the same directory, with a name in the form of a 7-character hash value.
```